src/utils/resolve.ts

```diff
@@ -125,14 +129,21 @@ function fullResolve(
     }
   }
 
+  const childContextHashKey = makeContextCacheKey(context)
+
   const sourceDir = path.dirname(sourceFile)
 
   if (prevSettings !== settings) {
     memoizedHash = stableHash(settings)
     prevSettings = settings
   }
 
-  const cacheKey = sourceDir + memoizedHash + modulePath
+  const cacheKey = [
+    sourceDir,
+    childContextHashKey,
+    memoizedHash,
+    modulePath,
+  ].join(',')
```

Comment:
`resolve` is a very hot path, and we need to squeeze any performance we can from this. We should manually concat the path using `+` instead of `Array#join`: https://fxtwitter.com/isukkaw/status/1925020329631735985

Reply:
Currently, this implementation creates a lot of one-use arrays, which creates GC pressure. No matter how fast the engine's GC is, it will never be faster than no GC at all.

Reply:
Already changed.